Source/Core/VideoCommon/VertexLoaderX64.cpp

```diff
@@ -187,13 +175,55 @@ int VertexLoaderX64::ReadVertex(OpArg data, u64 attribute, int format, int count
 			else
 				PSRLD(coords, 16);
 			break;
+		case FORMAT_FLOAT:
+			// Floats don't need to be scaled or converted,
+			// so we can just load/swap/store them directly
+			// and return early.
+			// (In SSSE3 we still need to store them.)
+			for (int i = 0; i < count_in; i++)
+			{
+				LoadAndSwap(32, scratch3, data);
+				MOV(32, dest, R(scratch3));
+				data.AddMemOffset(sizeof(float));
+				dest.AddMemOffset(sizeof(float));
+
+				// zfreeze
+				if (native_format == &m_native_vtx_decl.position)
+				{
+					if (cpu_info.bSSE4_1)
+					{
+						PINSRD(XMM0, R(scratch3), i);
```

Comment:
coords == XMM0
Should this and the next two be using the coords variable to make it less confusing what is happening?